modules/apps/transfer/keeper/grpc_query.go

```diff
@@ -112,13 +114,26 @@ func (k Keeper) DenomHash(c context.Context, req *types.QueryDenomHashRequest) (
 }
 
 // EscrowAddress implements the EscrowAddress gRPC method
-func (Keeper) EscrowAddress(c context.Context, req *types.QueryEscrowAddressRequest) (*types.QueryEscrowAddressResponse, error) {
+func (k Keeper) EscrowAddress(c context.Context, req *types.QueryEscrowAddressRequest) (*types.QueryEscrowAddressResponse, error) {
 	if req == nil {
 		return nil, status.Error(codes.InvalidArgument, "empty request")
 	}
 
 	addr := types.GetEscrowAddress(req.PortId, req.ChannelId)
 
+	if err := validategRPCRequest(req.PortId, req.ChannelId); err != nil {
+		return nil, err
+	}
+
+	ctx := sdk.UnwrapSDKContext(c)
+	found := k.channelKeeper.HasChannel(ctx, req.PortId, req.ChannelId)
+	if !found {
```

Comment:
This could be condensed but I don't really feel strongly about it...

```suggestion
	if !k.channelKeeper.HasChannel(ctx, req.PortId, req.ChannelId) {
```